modules/accounts/email.nix

```diff
@@ -6,6 +6,45 @@ let
 
   cfg = config.accounts.email;
 
+  gpgModule = types.submodule {
+    options = {
+      gpgKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "Path to your gpg key.";
+      };
+
+      onSend = mkOption {
+        type = types.enum [ "sign" "attach" "nothing" ];
+        default = "nothing";
+        description = "Method to communicate the signature.";
+      };
+    };
+  };
+
+  signatureModule = types.submodule {
+    options = {
+      content = mkOption {
```

Comment:
I think this is fair to include, but perhaps call it `text` instead of `content`? That would also allow the natural addition of a `html` option if need be in the future.